addons/website_form/models/models.py

```diff
@@ -49,4 +69,38 @@ class website_form_model_fields(models.Model):
     _name = 'ir.model.fields'
     _inherit = 'ir.model.fields'
 
-    website_form_blacklisted = fields.Boolean('Blacklisted in web forms', help='Blacklist this field for web forms')
+    def init(self, cr):
+        # set all existing unset website_form_blacklisted fields to ``true``
+        #  (so that we can use it as a whitelist rather than a blacklist)
+        cr.execute('UPDATE ir_model_fields'
+                   ' SET website_form_blacklisted=true'
+                   ' WHERE website_form_blacklisted IS NULL')
+        # add an SQL-level default value on website_form_blacklisted to that
+        # pure-SQL ir.model.field creations (e.g. in _field_create) generate
+        # the right default value for a whitelist (aka fields should be
+        # blacklisted by default)
+        cr.execute('ALTER TABLE ir_model_fields '
+                   ' ALTER COLUMN website_form_blacklisted SET DEFAULT true')
+
+    @api.model
+    def formbuilder_whitelist(self, model, fields):
+        """
+        :param str model: name of the model on which to whitelist fields
+        :param list(str) fields: list of fields to whitelist on the model
+        :return: nothing of import
+        """
+        # todo: check access rights yo
+        # the ORM only allows writing on custom fields and will trigger a
+        # registry reload once that's happened. We want to be able to
+        # whitelist non-custom fields and the registry reload absolutely
+        # isn't desirable, so go with a method and raw SQL
+        self.env.cr.execute(
+            "UPDATE ir_model_fields"
+            " SET website_form_blacklisted=false"
```

Comment:
> How to re-black list a base field?

Not currently possible, there's no case for it UI-wise at this point (the formbuilder can't do it because without a tracking model it doesn't know whether a field removed from one form is used by an other), it would require some sort of toggle button on the field object, which we could just as easily do with a server action.

> Shouldn't this be `SET website_form_blacklist = name NOT IN %s`

That wouldn't allow incremental whitelisting which is something we want for the formbuilder UI.